app/src/components/common/Table.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div class="table-container">
+    <table class="user-table">
+      <thead v-if="columns.length">
+        <tr>
+          <th
+            v-for="column in columns"
+            :key="column.key"
+          >
+            {{ getLabel(column) }}
+          </th>
+        </tr>
+      </thead>
+      <tbody v-if="data.length > 0">
+        <tr
+          v-for="item in data"
+          :key="item.id"
+        >
+          <td
+            v-for="column in columns"
+            :key="column.key"
+          >
+            <template v-if="column.slot">
+              <slot
+                :name="column.slot"
+                :item="item"
+              />
+            </template>
+            <template v-else>
+              {{ getEntry(column, item) }}
+            </template>
+          </td>
+        </tr>
+      </tbody>
+      <tbody v-else-if="columns.length > 0">
+        <tr>
+          <td :colspan="columns.length">
+            <p>Waiting for your data!</p>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script lang="ts">
+import { TableColumn } from '@/types'
+import { Vue, Component, Prop } from 'vue-property-decorator'
+
+@Component
+export default class Table extends Vue {
+  @Prop({ required: true }) columns!: TableColumn[]
+  @Prop({ required: true }) data!: Array<any>
+
+  getLabel (column: TableColumn): string {
+    return column.label ? column.label : ''
+  }
+
+  getEntry (column: TableColumn, item: any): string {
```

Comment:
I use the any type here as data for the table can be array of objects with any type and item is the one object of the input array.

Reply:
try to avoid `any` as much as possible. 
Maybe `{ [key: string]: string }` will work here? Or `{ [key: string]: string | number }`? At least we will show that `item` is an object.